src/components/NetworkDropdown.tsx

```diff
@@ -32,7 +32,14 @@ const Item = ({
 
 const NetworkDropdown = () => {
   const [networkConfig, setNetworkConfig] = useAtom(networkConfigAtom);
-
+  const specifiedNetworkName = new URLSearchParams(window.location.search).get('network');
+  if (specifiedNetworkName !== null && specifiedNetworkName in networkConfigs) {
+    setNetworkConfig((networkConfigs as never)[specifiedNetworkName]);
```

Comment:
Does this persist in localstorage? If it's in-memory it'll be wiped out when `window.location.replace` does a navigation.

Either way, better to use [pushState](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState) that a navigation.